flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -484,63 +484,23 @@ private DataStream<RowData> distributeDataStream(
         PartitionSpec partitionSpec,
         Schema iSchema,
         RowType flinkRowType) {
-      DistributionMode writeMode = flinkWriteConf.distributionMode();
+      DistributionMode writeMode =
+          flinkWriteConf.distributionMode(equalityFieldIds, equalityFieldColumns);
 
       LOG.info("Write distribution mode is '{}'", writeMode.modeName());
       switch (writeMode) {
         case NONE:
-          if (equalityFieldIds.isEmpty()) {
-            return input;
-          } else {
-            LOG.info("Distribute rows by equality fields, because there are equality fields set");
-            return input.keyBy(
-                new EqualityFieldKeySelector(iSchema, flinkRowType, equalityFieldIds));
-          }
+          return input;
 
         case HASH:
-          if (equalityFieldIds.isEmpty()) {
-            if (partitionSpec.isUnpartitioned()) {
-              LOG.warn(
-                  "Fallback to use 'none' distribution mode, because there are no equality fields set "
-                      + "and table is unpartitioned");
-              return input;
-            } else {
-              return input.keyBy(new PartitionKeySelector(partitionSpec, iSchema, flinkRowType));
-            }
-          } else {
-            if (partitionSpec.isUnpartitioned()) {
-              LOG.info(
-                  "Distribute rows by equality fields, because there are equality fields set "
-                      + "and table is unpartitioned");
-              return input.keyBy(
-                  new EqualityFieldKeySelector(iSchema, flinkRowType, equalityFieldIds));
-            } else {
-              for (PartitionField partitionField : partitionSpec.fields()) {
-                Preconditions.checkState(
-                    equalityFieldIds.contains(partitionField.sourceId()),
-                    "In 'hash' distribution mode with equality fields set, partition field '%s' "
-                        + "should be included in equality fields: '%s'",
-                    partitionField,
-                    equalityFieldColumns);
-              }
-              return input.keyBy(new PartitionKeySelector(partitionSpec, iSchema, flinkRowType));
-            }
-          }
-
-        case RANGE:
-          if (equalityFieldIds.isEmpty()) {
-            LOG.warn(
-                "Fallback to use 'none' distribution mode, because there are no equality fields set "
-                    + "and {}=range is not supported yet in flink",
-                WRITE_DISTRIBUTION_MODE);
-            return input;
-          } else {
+          if (partitionSpec.isUnpartitioned()) {
```

Comment:
is this correct? in the old logic, if it is `RANGE` and `equalityFieldIds` is not empty, we will use `EqualityFieldKeySelector`. that doesn't seem to be the case with this change.

Reply:
I did make a change here, sorry I forgot to add a comment.

Because for now, Range is not supported, if it's a non-partitioned table, it's still going to be EqualityFieldKeySelector, and for partitioned tables, there's no obvious benefit to using EqualityFieldKeySelector.

But I've removed that part of the logic. So let's skip it.